wrappers/src/object-store/src/object_store.hpp

```diff
@@ -253,6 +253,11 @@ namespace realm {
         };
         InvalidLinkingObjectsPropertyException(Type error_type, std::string const& object_type, Property const& property);
     };
+
+    class PrimaryKeyNotDeclaredException : public ObjectSchemaValidationException {
```

Comment:
Yes, that has been added to ObjectStore since the level we are currently using in dotnet (waiting on moving to  using it as a submodule.)